docs/_plugins/javadoc_tag.rb

```diff
@@ -0,0 +1,280 @@
+
+    if tag_name == "jdoc_package"
+      @is_package_ref = true
+    elsif tag_name == "jdoc_old"
+      @use_previous_api_version = true
+    end
+  end
+
+  def render(var_ctx)
+
+    artifact_name, @type_fqcn = JDocNamespaceDeclaration::parse_fqcn(@type_fqcn, var_ctx)
+
+    JavadocTag::diagnose(artifact_name, @type_fqcn, @is_package_ref)
+
+    # Expand FQCN of arguments
+    @member_suffix.gsub!(JDocNamespaceDeclaration::NAMESPACED_FQCN_REGEX) {|fqcn| JDocNamespaceDeclaration::parse_fqcn(fqcn, var_ctx)[1]}
+
+    visible_name = JavadocTag::get_visible_name(@opts, @type_fqcn, @member_suffix, @is_package_ref)
+
+    # Hack to reference the package summary
+    # Has to be done after finding the visible_name
+    if @is_package_ref
+      @type_fqcn = @type_fqcn + ".package-summary"
+    end
+
+    # Hardcode the artifact version instead of using "latest"
+    api_version = var_ctx["site.pmd." + (@use_previous_api_version ? "previous_version" : "version")]
+
+
+    markup_link(visible_name, doclink(artifact_name, api_version, @type_fqcn, @member_suffix))
+  end
+
+  private
+
+  def doclink(artifact, api_version, type_name, member_suffix)
+    "https://javadoc.io/page/net.sourceforge.pmd/#{artifact}/#{api_version}/#{type_name.gsub("\.", "/")}.html##{member_suffix}"
+  end
+
+  def markup_link(rname, link)
+    "[`#{rname}`](#{link})"
+  end
+
+
+  def self.get_visible_name(opts, type_fqcn, member_suffix, is_package_ref)
+
+    # method or field
+    if member_suffix && Regexp.new('(\w+)(' + ARGUMENTS_REGEX.source + ")?") =~ member_suffix
+
+      suffix = $1 # method or field name
+
+      if opts.show_args? && $2 && !$2.empty? # is method
+
+        args = ($3 || "").split(",").map {|a| a.gsub(/\w+\./, "").strip} # map to simple names
+
+        suffix = "#{suffix}(#{args.join(", ")})"
+      end
+
+      visible_name = if opts.show_fqcn?
+                       type_fqcn + "#" + suffix
+                     elsif opts.is_double_bang? || opts.show_class?
+                       type_fqcn.split("\.").last + "#" + suffix # type simple name
+                     else
+                       suffix # just method name + possibly args
+                     end
+
+      return visible_name
+    end
+
+    # else package or type, for packages the FQCN_OPTION is present
+
+    if is_package_ref || opts.show_fqcn? || opts.is_double_bang?
+      type_fqcn
+    else
+      type_fqcn.split("\.").last # type simple name
+    end
+  end
+
+  BASE_PMD_DIR = File.join(File.expand_path(File.dirname(__FILE__)), "..", "..")
+
+  def self.diagnose(artifact_id, fqcn, expect_package)
+    resolved_type = JavadocTag::fqcn_type(artifact_id, fqcn)
+
+    tag_name= expect_package ? "jdoc_package" : "jdoc"
+
+    if resolved_type == :package && !expect_package
+      warn "\e[33;1m#{tag_name} generated link to #{fqcn}, but it was found to be a package name. Did you mean to use jdoc_package instead of jdoc?\e[0m"
```

Comment:
Yeah they would only be noticeable on on local builds for now. I originally used `fail` instead of `warn`, which *could* make the travis build noticeably fail (although idk). I backed off just because I'd rather avoid "false positives" where the build fails on correct input --though arguably we could start by using `fail` until an unfixable shortcoming is found